src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -303,221 +525,375 @@ else if (genotype.isCalled() && genotype.isHomRef()) {
-                    }
+            //note that subsetting alleles can increase GQ, e.g. with one confident reference allele and a deletion allele that's either 4 or 5 bases long
+            builder.genotypes(gc).alleles(newAlleleSetUntrimmed);
+            //if deletions are dropped, alleles may need trimming
+            final VariantContext newTrimmedAllelesVC = GATKVariantContextUtils.trimAlleles(builder.make(), false, true);
+            builder = new VariantContextBuilder(newTrimmedAllelesVC);
+            //save indices of new alleles for annotation processing
+            relevantIndices = newAlleleSetUntrimmed.stream().mapToInt(a -> variant.getAlleles().indexOf(a)).toArray();
+            final int refBlockDepth;
+            if (variant.hasAttribute(VCFConstants.DEPTH_KEY)) {  //prefer INFO depth because HaplotypeCaller GVCF block code uses all reads, not just informative
+                refBlockDepth = variant.getAttributeAsInt(VCFConstants.DEPTH_KEY, 0);
+            } else if (genotype.hasDP()) {
+                refBlockDepth = genotype.getDP();
+            } else {
+                refBlockDepth = 0;
+            }
+            addRefBlockIfNecessary(variant, allelesToDrop, newTrimmedAllelesVC, refBlockDepth);
+        }
+
+        final VariantContext updatedAllelesVC = builder.make();
+        final Genotype updatedAllelesGenotype = updatedAllelesVC.getGenotype(0);
+
+        //remove any AD reads for the non-ref
+        final ArrayList<Genotype> genotypesArray = removeNonRefADs(updatedAllelesGenotype, updatedAllelesVC.getAlleleIndex(Allele.NON_REF_ALLELE));
+        builder.genotypes(genotypesArray);
+
+        composeUpdatedAnnotations(variant, attrMap, relevantIndices, updatedAllelesVC);
+
+        return builder.attributes(attrMap).unfiltered().make();
+    }
+
+    /**
+     * Update annotations if alleles were subset and add annotations specific to ReblockGVCF, like QUALapprox and RAW_GT_COUNT
+     * @param variant   variant context with full annotation data
+     * @param attrMap   annotation map to modify with new annotations
+     * @param relevantIndices   indexes of alleles in updatedAllelesVC with respect to variant
+     * @param updatedAllelesVC  variant context with final set of alleles
+     */
+    private void composeUpdatedAnnotations(final VariantContext variant, final Map<String, Object> attrMap,
+                                           final int[] relevantIndices, final VariantContext updatedAllelesVC) {
+        updateMQAnnotations(variant, attrMap);
+
+        final boolean allelesNeedSubsetting = relevantIndices.length < variant.getNAlleles();
+        copyInfoAnnotations(variant, attrMap, allelesNeedSubsetting, relevantIndices);
+
+        //generate qual annotations after we potentially drop alleles
+        final Genotype updatedAllelesGenotype = updatedAllelesVC.getGenotype(0);
+        if (doQualApprox) {
+            //TODO: update to support DRAGEN posteriors
```

Comment:
Isn't this code partially for the purpose of reblocking DRAGEN GVCFs? These lines should probably link to #6269 since this handling will be related to what decision we make there.